client/trino-client/src/main/java/io/trino/client/auth/external/KnownToken.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.client.auth.external;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+public interface KnownToken
+{
+    Optional<Token> getToken();
+
+    void setupToken(Supplier<Token> tokenSource);
+
+    static KnownToken local()
```

Comment:
what local means? Thread local? Or just JDBC connection local?

Reply:
jdbc connection local in this case. That's the best names that I came up with :)